client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Internal dependencies
+ */
+import { MEDIA_REQUEST } from 'state/action-types';
+import { isRequestingMedia } from 'state/selectors';
+import { failMediaRequest, receiveMedia, requestingMedia } from 'state/media/actions';
+import wpcom from 'lib/wp';
+import debug from 'debug';
+
+/**
+ * Module variables
+ */
+const log = debug( 'calypso:middleware-media' );
+
+/**
+ * Issues an API request to fetch media for a site and query.
+ *
+ * @param  {Object}  store  Redux store
+ * @param  {Object}  action Action object
+ * @return {Promise}        Promise
+ */
+export function requestMedia( { dispatch, getState }, { siteId, query } ) {
+	if ( ! isRequestingMedia( getState(), siteId, query ) ) {
+		dispatch( requestingMedia( siteId, query ) );
+
+		log( 'Request media for site %d using query %o', siteId, query );
+
+		return wpcom.site( siteId ).mediaList( query ).then( ( { media, found } ) => {
+			dispatch( receiveMedia( siteId, media, found, query ) );
+		} ).catch( () => {
+			dispatch( failMediaRequest( siteId, query ) );
+		} );
```

Comment:
Although it's a personal style request and I don't think we have this in our guidelines, would you be interested in breaking this into multiple lines? at a minimum so that `.then` and `.catch` start on their own lines? (You can feel free to say "no" to this)

```js
return wpcom
	.site( siteId )
	.mediaList( query )
	.then( ( { media, found } ) => dispatch( receiveMedia( siteId, media, found, query ) ) )
	.catch( () => dispatch( failMediaRequest( siteId, query ) ) );
```